src/commands/indexer/index.ts

```diff
@@ -533,15 +535,38 @@ export default class Indexer extends HealthCheck {
             const crossChainMessageSentEvent: CrossChainMessageSentEvent | null = this.bloomFilters[
               type
             ]!.bloomEvent.decode<CrossChainMessageSentEvent>(type, interestingTransaction.log!)
-            if (crossChainMessageSentEvent !== null) {
-              // should optimize SQS logic to not make additional calls since all data is already digested and parsed here
-              await sqsHandleBridgeEvent.call(
-                this,
-                this.networkMonitor,
-                interestingTransaction.transaction,
-                job.network,
-                tags,
-              )
+
+            // TODO: Check the various types of cross chain messages
+            let crossChainMessageType: CrossChainMessageType | null = CrossChainMessageType.ERC721
+            if (interestingTransaction.allLogs) {
+              // Make sure we have logs
+              for (const log of interestingTransaction.allLogs) {
```

Comment:
@alexanderattar so i know this is still a draft but i would prefer this be a function and not have this get too nested. I was imagining something like this:


```
// previous code above
let crossChainMessageType: CrossChainMessageType = detectCrossChainType(interestingTransaction.allLogs)

if (crossChainMessageSentEvent !== null) {
  // should optimize SQS logic to not make additional calls since all data is already digested and parsed here
  await sqsHandleBridgeEvent.call(
    this,
    this.networkMonitor,
    interestingTransaction.transaction,
    job.network,
    crossChainMessageType,
    tags,
  )
}

async detectCrossChainType(allLogs) {

  const containsCollection = allLogs.filter( --- )
  const containsNFT = allLogs.filter( --- )

  if (containsCollection.length > == 1 ) {
    return  CrossChainMessageType.CONTRACT
  } else if (containsCollection.length > == 1) {
    return CrossChainMessageType.NFT
  } else {
    // Maybe throw error or just return unknown type and handle outside of fun
    return CrossChainMessageType.UNKNOWN
  } 
}
```

The above reads betters and does not have the logic of the loop linked to the sqs function. I would also suggest using `filter` or maybe `include` to detect what is contained in the `allLogs` array. Feel free to change function or variable names or make more organized.

Reply:
Yeah something like this is what I had in mind

Reply:
The only thing is that I don't think we should loop over allLogs multiple times. We can just do that once